src/custom_component_renderer/dom_manipulator.rs

```diff
@@ -0,0 +1,147 @@
+use super::error::RendererError;
+use crate::custom_component_renderer::error::Result;
+use ego_tree::{NodeId, NodeMut, NodeRef, Tree};
+use markup5ever::namespace_url;
+use markup5ever::{ns, Attribute, LocalName, QualName};
+use scraper::node::{Element, Text};
+use scraper::Node;
+
+pub struct NodeManipulator<'a> {
```

Comment:
When I saw this struct used in the other file, I had expected it to come from a library. Why are we writing all this code? 🙂